src/core/reaction_methods/tests/ReactionAlgorithm_test.cpp

```diff
@@ -230,6 +238,68 @@ BOOST_AUTO_TEST_CASE(ReactionAlgorithm_test) {
     remove_particle(1);
   }
 
+  // checks that the system energy is updated after a succesful reaction
+
+  {
+
+// Create an instance of the Reaction ensemble class
+#include "reaction_methods/ReactionEnsemble.hpp"
+
+    class ReactionEnsembleTest : public ReactionEnsemble {
+    public:
+      using ReactionEnsemble::generic_oneway_reaction;
+      using ReactionEnsemble::ReactionEnsemble;
+    };
+
+    // Setup the Reaction ensamble
+    ReactionEnsembleTest test_reaction(42);
+    test_reaction.volume = 1.;
+    test_reaction.kT = 1.;
+    test_reaction.exclusion_radius = 0;
+
+    // Create a generic identity exchange reaction  D <-> E
+
+    int const type_D = 0;
+    int const type_E = 1;
+
+    test_reaction.charges_of_types[0] = 0;
+    test_reaction.charges_of_types[1] = 0;
+
+    float Gamma = INFINITY; // Reaction completly shifted to product creation
+    ReactionMethods::SingleReaction reaction(Gamma, {type_D}, {1}, {type_E},
+                                             {1});
```

Comment:
Variable names start with a lowercase to distinguish them from struct/class names. We use `double` for floating-point values, except in CUDA code. We don't use `INFINITY` nor `std::numeric_limits<double>::infinity()` because infinity values are ignored when ESPResSo is compiled in fast-math mode. Would 10^100 be enough? I thought about `std::numeric_limits<double>::max()`, but then if any value is added to it when calculating the probability, there is a risk of overflow.

```suggestion
    auto const gamma = 1e100; // Reaction completely shifted to product creation
    ReactionMethods::SingleReaction reaction(gamma, {type_D}, {1}, {type_E},
                                             {1});
```